src/blocks/remote-data-container/components/item-list/ItemList.tsx

```diff
@@ -1,135 +1,133 @@
-									<img alt={ ( item.image_alt as string ) ?? '' } src={ item[ field ] as string } />
-								);
-						  }
-						: undefined,
-				enableSorting: field !== media,
-			};
-		} );
-
-		return { fields: fieldObject, tableFields: getFields, titleField: title, mediaField: media };
-	}, [ availableBindings, data ] );
+	const fieldNames: string[] = Array.from(
+		new Set(
+			data
+				?.flatMap( item => Object.keys( item ) )
+				.filter(
+					key => key in availableBindings && availableBindings[ key ]?.type !== 'id' // filter out ID fields to hide from table
+				)
+		)
+	);
+
+	// Find title field from availableBindings by checking type
+	const titleField = Object.entries( availableBindings ).find(
+		( [ _, binding ] ) => binding.type === 'string' && binding.name.toLowerCase() === 'title'
+	)?.[ 0 ];
+
+	// Find media field from availableBindings by checking type
+	const mediaField = Object.entries( availableBindings ).find(
+		( [ _, binding ] ) => binding.type === 'image_url'
+	)?.[ 0 ];
+
+	const fields = fieldNames.map( field => {
+		return {
+			id: field,
+			label: availableBindings[ field ]?.name ?? field,
+			enableGlobalSearch: true,
+			getValue: ( { item }: { item: RemoteDataResult } ) => item[ field ] as string,
+			render:
+				field === mediaField
+					? ( { item }: { item: RemoteDataResult } ) => {
+							return (
+								<img alt={ ( item.image_alt as string ) ?? '' } src={ item[ field ] as string } />
+							);
+					  }
+					: undefined,
+			enableSorting: field !== mediaField,
+		};
+	} );
+
+	// hide media and title fields from table view if defined to avoid duplication
+	const tableFields = fieldNames.filter( field => field !== mediaField && field !== titleField );
 
 	const [ view, setView ] = useState< View >( {
 		type: 'table' as const,
-		perPage: 8,
-		page: 1,
-		search: '',
-		fields: [],
+		perPage: perPage ?? data.length,
+		page,
+		search: searchInput,
+		fields: tableFields,
 		filters: [],
 		layout: {},
 		titleField,
 		mediaField,
 	} );
 
+	function onChangeView( newView: View ) {
+		setPage( newView.page ?? 1 );
+		setSearchInput( newView.search ?? '' );
+
+		setView( newView );
+	}
+
 	const defaultLayouts = mediaField
 		? {
 				table: {},
 				grid: {},
 		  }
 		: { table: {} };
 
-	// this prevents just an empty table rendering
-	useEffect( () => {
-		if ( tableFields.length > 0 ) {
-			setView( prevView => ( {
-				...prevView,
-				// hide media and title fields from table view if defined to avoid duplication
-				fields: tableFields.filter( field => field !== mediaField && field !== titleField ),
-			} ) );
-		}
-	}, [ mediaField, tableFields, titleField ] );
-
-	useEffect( () => {
-		if ( view.search !== searchTerms ) {
-			setSearchTerms( view.search ?? '' );
-		}
-	}, [ view, searchTerms ] );
-
-	// filter, sort and paginate data
-	const { data: filteredData, paginationInfo } = useMemo( () => {
-		return filterSortAndPaginate( data ?? [], view, fields );
-	}, [ data, view ] );
```

Comment:
@chriszarate are we removing the search for upcoming changes in #337? 

I noticed because our integrated sources don't have pagination, since now it'll fallback to `data.length` if pagination isn't explicitly set: https://github.com/Automattic/remote-data-blocks/blob/eed54e2dedff4db8b5610738dbfec724f7605f56/src/blocks/remote-data-container/components/item-list/ItemList.tsx#L107-L108
https://github.com/Automattic/remote-data-blocks/blob/eed54e2dedff4db8b5610738dbfec724f7605f56/src/blocks/remote-data-container/components/item-list/ItemList.tsx#L154-L157

Reply:
@brookewp The reasons for removing search (and filtering) are summarized here:

https://wp.me/pfvsEV-tn#behavior-change-for-searching-sorting-and-filtering-in-dataviews

After this PR is approved, I'll work on implementing pagination for our other integrations.